crates/scroll/evm/src/config.rs

```diff
@@ -0,0 +1,329 @@
+use reth_chainspec::Head;
+use reth_evm::{ConfigureEvm, ConfigureEvmEnv, NextBlockEnvAttributes};
+use reth_primitives::{transaction::FillTxEnv, TransactionSigned};
+use reth_revm::{inspector_handle_register, Database, Evm, GetInspector, TxEnv};
+use reth_scroll_chainspec::{ScrollChainConfig, ScrollChainSpec};
+use reth_scroll_forks::ScrollHardfork;
+use revm::{
+    precompile::{Address, Bytes},
+    primitives::{
+        AnalysisKind, BlockEnv, CfgEnv, CfgEnvWithHandlerCfg, Env, HandlerCfg, SpecId, U256,
+    },
+    EvmBuilder,
+};
+use std::{convert::Infallible, sync::Arc};
+
+/// Scroll EVM configuration.
+#[derive(Clone, Debug)]
+pub struct ScrollEvmConfig {
+    /// The chain spec for Scroll.
+    chain_spec: Arc<ScrollChainSpec>,
+    /// Additional Scroll configuration.
+    scroll_config: ScrollChainConfig,
```

Comment:
question: Should this be `Arc< ScrollChainConfig>`?

Reply:
I think the idea of the `Arc` for the `ScrollChainSpec` is that cloning it might incur a high cost if the genesis is large. In the case of the `ScrollChainConfig`, I think the cost for cloning is a lot more limited so I decided to keep it as is.